lib/umwelt/client/clone.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Umwelt.Client.Clone do
+  @moduledoc "Clone main process"
+
+  use GenServer
+  require Logger
+
+  alias Umwelt.Client
+
+  def start_link(_),
+    do:
+      GenServer.start_link(
+        __MODULE__,
+        %{phase_id: nil, port: nil},
+        name: __MODULE__
+      )
+
+  def init(state), do: {:ok, state}
+
+  def handle_cast({:pull, params}, _state) do
+    send(self(), :start_pulling)
+
+    {:noreply, params}
+  end
+
+  def handle_info(:start_pulling, state) do
+    case Client.Request.fetch_modules(state) do
+      {:ok, modules} ->
+        Logger.info("Fetching modules: #{inspect(Map.keys(modules))}")
+        modules |> Client.Agent.add_modules()
+
+      {:error, reason} ->
+        Logger.error("Failed to fetch modules: #{inspect(reason)}. Stopping...")
+        Supervisor.stop(Client.Supervisor)
+    end
+
+    send(self(), :spawn_fetchers)
```

Comment:
Why do we ever need another message here? What would be wrong with inlining `spawn_fetchers` code here?

Reply:
It was an attempt to test it step by step. Joined them.